sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs

```diff
@@ -0,0 +1,167 @@
+
+            // Create StorageSyncService
+            _storageSyncServiceResource = (await _resourceGroup.GetStorageSyncServices().CreateOrUpdateAsync(WaitUntil.Completed, _storageSyncServiceName, _storageSyncServiceCreateOrUpdateContent)).Value;
+            Assert.NotNull(_storageSyncServiceResource);
+            StorageSyncManagementTestUtilities.VerifyStorageSyncServiceProperties(_storageSyncServiceResource, true);
+
+            // Create StorageSyncGroup
+            _storageSyncGroupResource = (await _storageSyncServiceResource.GetStorageSyncGroups().CreateOrUpdateAsync(WaitUntil.Completed, _syncGroupName, _storageSyncGroupCreateOrUpdateContent)).Value;
+            Assert.NotNull(_storageSyncGroupResource);
+            StorageSyncManagementTestUtilities.VerifySyncGroupProperties(_storageSyncGroupResource, true);
+        }
+
+        [TearDown]
+        public async Task DeleteStorageSyncResources()
+        {
+            var storageSyncServiseExists = (await _resourceGroup.GetStorageSyncServices().ExistsAsync(_storageSyncServiceName)).Value;
+            if (storageSyncServiseExists)
+            {
+                var storageSyncService = (await _resourceGroup.GetStorageSyncServiceAsync(_storageSyncServiceName)).Value;
+                if ((await storageSyncService.GetStorageSyncGroups().ExistsAsync(_syncGroupName)).Value)
+                {
+                    var syncGroupResource = (await storageSyncService.GetStorageSyncGroupAsync(_syncGroupName)).Value;
+                    if ((await syncGroupResource.GetCloudEndpoints().ExistsAsync(_cloudEndpointName)).Value)
+                    {
+                        var cloudEndpointResource = (await syncGroupResource.GetCloudEndpointAsync(_cloudEndpointName)).Value;
+                        await cloudEndpointResource.DeleteAsync(WaitUntil.Completed);
+                    }
+                    await syncGroupResource.DeleteAsync(WaitUntil.Completed);
+                }
+                await storageSyncService.DeleteAsync(WaitUntil.Completed);
+            }
+        }
+
+        [Test]
+        [RecordedTest]
+        public async Task CloudEndpointCreateTest()
+        {
+            // Create CloudEndpoint
+            CloudEndpointResource cloudEndpointResource = (await _storageSyncGroupResource.GetCloudEndpoints().CreateOrUpdateAsync(WaitUntil.Completed, _cloudEndpointName, _cloudEndpointCreateOrUpdateContent)).Value;
+            Assert.NotNull(cloudEndpointResource);
+            StorageSyncManagementTestUtilities.VerifyCloudEndpointProperties(cloudEndpointResource, true);
+        }
+
+        [Test]
+        [RecordedTest]
+        public async Task CloudEndpointGetTest()
+        {
+            // Create CloudEndpoint
+            CloudEndpointResource cloudEndpointResource = (await _storageSyncGroupResource.GetCloudEndpoints().CreateOrUpdateAsync(WaitUntil.Completed, _cloudEndpointName, _cloudEndpointCreateOrUpdateContent)).Value;
```

Comment:
Option 1 :  Get as is, Check HasValue using Assert.  Use .Value
Option 2 : Use GetValueOrDefault()